### DISCUSSION TITLE
Backup rotation

### DISCUSSION THREAD

**Discussion by loeffelpan** [nextcloud/all-in-one#1674] [Questions] [answered]
I'm wondering about rotating the backups with integrated borg-solution.
Did I miss any configuration parameter to define maximum number or size of backups to keep?

Destination of my backup is a usb-stick with less capacity and I would like to keep e. g. only 30 days of backup.

**Comment by szaimen** [comment]
Hi, see https://github.com/nextcloud/all-in-one/discussions/1546#discussioncomment-4449021

And no, I dont want to make this configurable. However you can add your own prune script from the host via cronjob if you really need to. (Install borg, create a script with a borg prune command and create a cronjob).

**Reply by loeffelpan** [reply]
Thanks. I did a search but did not find that discussion.
I like that prune settings (7d, 4w and 6m) and going to keep them.
Hopefully my usb-stick has enough capacity. If not I will use a bigger one.

**Reply by szaimen** [reply]
> I did a search but did not find that discussion.

I've created https://github.com/nextcloud/all-in-one/discussions/1675 now which hopefully makes it easier to find.

**Comment by loeffelpan** [comment]
I have an additional question to this.
Yesterday I noticed that that `borg prune --keep-daily=1` will skip exactly one backup from being deleted. Which is the one last with additional host mount that I configured. `--keep-within=1d` will keep all backups from that days. This is fine.
But I think that your `--keep-weekly=4` and `--keep-monthly=6` will only keep two weekly nextcloud-backups and two weekly backups from additional host mounts as they are configured.

Can you please check that? I don't have an instance old enough to have weekly oder monthly backups.
Maybe you have to change your prune options if additional host mount are configured for backup to ensure that four weekly and six monthly **nextcloud**-backups are kept.

**Reply by szaimen** [reply]
Hi, this works fine here. I suppose you have missed the prefix: https://github.com/nextcloud/all-in-one/blob/756511f16afaac36f0d78d80891ec81c656878fa/Containers/borgbackup/backupscript.sh#L183 https://github.com/nextcloud/all-in-one/blob/756511f16afaac36f0d78d80891ec81c656878fa/Containers/borgbackup/backupscript.sh#L154 https://github.com/nextcloud/all-in-one/blob/756511f16afaac36f0d78d80891ec81c656878fa/Containers/borgbackup/backupscript.sh#L210

**Reply by loeffelpan** [reply]
Yeah. Thanks for clarification.

**Comment by szaimen** [comment, marked as answer]
Hi, see https://github.com/nextcloud/all-in-one/discussions/2375
